static/app/components/profiling/profilingOnboardingSidebar.tsx

```diff
@@ -49,7 +51,37 @@ const PROFILING_ONBOARDING_STEPS = [
   ProductSolution.PROFILING,
 ];
 
-export function ProfilingOnboardingSidebar(props: CommonSidebarProps) {
+export function useProfilingOnboardingDrawer() {
+  const organization = useOrganization();
+  const currentPanel = useLegacyStore(SidebarPanelStore);
+  const isActive = currentPanel === SidebarPanelKey.PROFILING_ONBOARDING;
+  const hasProjectAccess = organization.access.includes('project:read');
+
+  const {openDrawer} = useDrawer();
+
+  useEffect(() => {
```

Comment:
```suggestion
  useLayoutEffect(() => {
```

Reply:
Nit, but using useLayoutEffect here might mitigate some odd flickering in the future.